src/components/PopupWithForm.js

```diff
@@ -16,9 +17,13 @@ class PopupWithForm extends Popup {
     return formValues;
   }
 
-  submit(e) {
+  submit(e) { // handles all submitting stuff
     e.preventDefault();
+    const buttonLabel = this._popupElement.querySelector(".modal__save").innerText;
+    this._popupElement.querySelector(".modal__save").innerText = "Saving....";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```